package.json

```diff
@@ -89,6 +89,36 @@
                     "default": "off",
                     "description": "Traces the communication between VSCode and the YML language server."
                 },
+                "yseopml.documentFormat": {
+                    "type": "object",
+                    "default": {
+                        "enableDocumentFormat": "yes",
+                        "semicolonWhenPossible": true,
+                        "spaceBetweenKeywordAndParenthesis": false
+                    },
+                    "description": "Settings related to document format",
+                    "properties": {
+                        "enableDocumentFormat": {
+                            "type": "string",
```

Comment:
Really? Booleans are not just checkboxes or something like this in the GUI version?
But then why stick to booleans for the semicolon thingy and the… other thing with a long name?

Reply:
Note: Strings may allow further evolution, like “onlyIfBlahBlah”. Not sure it would be trivial to switch, on extension update, from a boolean to something like this xd

Reply:
(I mean, two states vs. more than that)

Reply:
> Really? Booleans are not just checkboxes or something like this in the GUI version?

This is true only for first-level options. Here we are modifying the attributes of an object.

> But then why stick to booleans for the semicolon thingy and the… other thing with a long name?

I didn't like the yes-or-no thingy in the first place.

> Note: Strings may allow further evolution, like “onlyIfBlahBlah”.
(I mean, two states vs. more than that)

Exactly! I was thinking about it too.

> Not sure it would be trivial to switch, on extension update, from a boolean to something like this xd

I don't think so. We'll have to use deprecation stuff. :sweat:

Reply:
I still don’t quite get why you don’t use the same thing for all three options, then xd
Sounds like since you don’t completely like yes/no, you only use it for one out of the three. But then you agree on the fact that yes/no is nicer for evolutions. :exploding_head:

Reply:
I was focused on UI instead of UI _and_ UX. I'll change it to yes/no everywhere.